crates/engine/tree/src/tree/mod.rs

```diff
@@ -725,11 +725,15 @@ where
     ///   extension of the canonical chain.
     /// * walking back from the current head to verify that the target hash is not already part of
     ///   the canonical chain.
-    fn is_fork(&self, target_hash: B256) -> ProviderResult<bool> {
+    fn is_fork(&self, target_header: SealedHeader<N::BlockHeader>) -> ProviderResult<bool> {
```

Comment:
is the header required as an arg now because we might be checking that the header is a fork block before it's in the tree state (_and_ before it's in the db?)

if so can we add fn docs for this